### PR TITLE
Shadowling, Changeling, and Vision Fixes/Changes

### PR DESCRIPTION
**Pull request by TheDZD**
- Ports most of tgstation/-tg-station#8291 (does not port the carbon.dm changes, as we don't have a flash_eyes proc.)
- Ports tgstation/-tg-station#11231
- Ports tgstation/-tg-station#11784
- Creates a new vision_override datum system, allows for constant-effect vision changes despite shitty vision handling in human Life().
 - Only actually overrides see_invisible and weakeyes, vision flags stack with things like mesons, and darkvision distance picks the highest from the player's current see_in_dark (including if affected by NVGs or the like), the datum's see_in_dark, and the players' species' darksight.
 - Planning to expand this in the future to handle more than just toggleable vision-based abilities.
- Enables shadowling thralls to have toggleable night vision by using vision_override.
- Allows for shadowlings to toggle their thermal+night vision sight so they stop dying because they don't know where light is coming from.
 - Un-hatched shadowlings get this ability to use thermal sight as well now, to shorten the huge power gap between hatched and non-hatched lings a bit, as a bonus, they get thermal vision without becoming horribly weak to bright flashes. Hatched lings still have thermal sight, but it is now also toggleable, bright flashes are going to wreck them regardless of whether or not it is on.

Fixes https://github.com/ParadiseSS13/Paradise/issues/1375.

### CODE REVIEW

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 1]
```diff
@@ -36,9 +36,17 @@
 				ear_safety++
 
 //Flash
+	var/mob/living/carbon/human/H = M
+	var/obj/item/organ/eyes/E = H.internal_organs_by_name["eyes"]
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/grenades/flashbang.dm]: not sane, it was behind an ishuman check for a reason

File: code/datums/vision_override.dm [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/datum/vision_override
+	var/name = "vision override"
+	var/see_in_dark = 0
+	var/see_invisible = 0
+	var/light_sensetive = 0
```
Comment (Markolie) [inline code/datums/vision_override.dm]: sensitive

File: code/game/objects/items/weapons/grenades/flashbang.dm (hunk 1 above)
Comment (TheDZD) [inline code/game/objects/items/weapons/grenades/flashbang.dm]: Whoops, missed that when moving this section up.

File: code/datums/vision_override.dm (hunk 2 above)
Comment (TheDZD) [inline code/datums/vision_override.dm]: I actually don't know how I managed to consistently misspell this.

**Comment by TheDZD** [comment]
I hate merge commits, but Git was being particularly annoying and telling me I still had the merge conflict that I fixed in the commit prior to it.

**Comment by TheDZD** [comment]
Tested this, should be merge-ready once Travis agrees.

**Pull request merged**